inference-engine/src/transformations/src/transformations/common_optimizations/strides_optimization.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+#include <transformations/common_optimizations/strides_optimization.hpp>
+#include <transformations/rt_info/strides_property.hpp>
+#include <transformations/utils/utils.hpp>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/variant.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::StridesOptimization, "StridesOptimization", 0);
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvStridesPropagation, "ConvStridesPropagation", 0);
+
+static bool can_propagate_conv_stride(const std::shared_ptr<ngraph::Node>& conv) {
+    const auto& kernel_shape = conv->input_value(1).get_shape();
+    return std::all_of(kernel_shape.begin() + 2, kernel_shape.end(), [] (size_t s) -> bool { return s == 1; });
+}
+
+static std::tuple<ngraph::Strides, bool> check_next_ops(const std::vector<ngraph::Input<ngraph::Node>>& next_ops) {
+    std::vector<ngraph::Strides> strides;
+    for (const auto& op : next_ops) {
+        if (!has_strides_prop(op)) {
+            return std::make_tuple(ngraph::Strides{}, false);
+        }
+        strides.push_back(get_strides_prop(op));
+    }
+    bool all_ops_are_valid = std::all_of(strides.begin(), strides.end(),
+                                         [&strides] (const ngraph::Strides& s) -> bool {
+                                             bool all_ones = std::all_of(s.begin(), s.end(), [] (size_t i) -> bool { return i == 1; });
+                                             return s == strides[0] && !all_ones;
+                                        });
+    return std::make_tuple(strides[0], all_ops_are_valid);
+}
+
+static void insert_pooling(const ngraph::Output<ngraph::Node>& first, ngraph::Input<ngraph::Node>& second, const ngraph::Strides& strides) {
+    auto first_node = first.get_node_shared_ptr();
+    auto rank = first.get_partial_shape().rank();
+    if (rank.is_static() && static_cast<size_t>(rank.get_length()) < strides.size() + 2) {
+        size_t diff = strides.size() + 2 - static_cast<size_t>(rank.get_length());
+        auto ones = ngraph::opset7::Constant::create(ngraph::element::i64, ngraph::Shape{diff}, std::vector<int64_t>(diff, 1));
+        auto current_shape = std::make_shared<ngraph::opset7::ShapeOf>(first);
+        auto new_shape = std::make_shared<ngraph::opset7::Concat>(ngraph::OutputVector{ones, current_shape}, 0);
+        first_node = std::make_shared<ngraph::opset7::Reshape>(first_node, new_shape, false);
```

Comment:
If you broadcast input shape with ones you need to reshape it back after pooling (for example you can change eltwise output shape if you make rank bigger just because of broadasting).

Reply:
Please add necessary test to check different broadcasting scenarious.